src/Key.php

```diff
@@ -5,18 +5,132 @@
 class Key
 {
     /** @var string */
-    public $id;
+    private $id;
 
     /** @var string */
-    public $secret;
+    private $secret;
+
+    /** @var string */
+    private $certificate;
+
+    /** @var string */
+    private $privateKey;
+
+    /** @var string */
+    private $type;
 
     /**
      * @param string $id
      * @param string $secret
      */
-    public function __construct($id, $secret)
+    public function __construct($id, $item)
     {
         $this->id = $id;
-        $this->secret = $secret;
+        $certificate = $this->getX509Certificate($item);
+        $privateKey = $this->getRSAPrivateKey($item);
+        if (($certificate || $privateKey)) {
+            $this->type = 'rsa';
+            if ($privateKey) {
+                $this->privateKey = $privateKey;
+            }
+            if ($certificate) {
+                $this->certificate = openssl_x509_read($certificate);
+            }
+            if ($certificate && $privateKey) {
+                if (!openssl_x509_check_private_key(
+                        $this->certificate,
+                        $this->privateKey
```

Comment:
code style - just inline it, there's no benefit for breaking this up

Reply:
I'm trying to keep the width below 80 chars, and for me at least breaking this up helps me visualise the logic blocks.